core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3230,6 +3230,8 @@ private void setAttributeInternal(RpcContext rpcContext, LockedInodePath inodePa
     long opTimeMs = System.currentTimeMillis();
     if (context.getOptions().getRecursive() && targetInode.isDirectory()) {
       try (LockedInodePathList descendants = mInodeTree.getDescendants(inodePath)) {
+        LOG.debug("Set attribute for {} descendants under path {}.",
+            descendants.getInodePathList().size(), inodePath.getUri());
```

Comment:
Is `getInodePathList()` expensive? If it is, then we should avoid doing this computation if debug is disabled.